paper_framework_sources/p1897_initial_executor_algorithms_for_cpp23.md

```diff
@@ -25,56 +35,71 @@ toc: false
+In recent discussions we have converged to an understanding of the `submit` operation on a `sender` and its more fundamental primitives `connect` and `start` supporting general interoperation between algorithms, and algorithm customization giving us full flexibility to optimize, to offload and to avoid synchronization in chains of mutually compatible algorithm customizations.
 
-As a starting point, in [@P0443R11] we only include a `bulk_execute` algorithm, that satisfies the core requirement we planned with [@P0443R11] to provide scalar and bulk execution.
+As a starting point, in [@P0443R13] we only include a `bulk_execute` algorithm, that satisfies the core requirement to provide scalar and bulk execution.
 To make the C++23 solution completely practical, we should extend the set of algorithms, however.
 This paper suggests an expanded initial set that enables early useful work chains.
 This set is intended to act as a discussion focus for us to discuss one by one, and to analyze the finer constraints of the wording to make sure we do not over-constrain the design.
 
+This paper does not attempt to propose the mapping of the C++20 parallel algorithms into an asynchronous environment.
+Once we have basic primitives, we can describe default implementations of the parallel algorithms, as well as `std::async`, in terms of these.
+
 In the long run we expect to have a much wider set of algorithms, potentially covering the full set in the current C++20 parallel algorithms.
 The precise customization of these algorithms is open to discussion: they may be individually customized and individually defaulted, or they may be optionally individually customized but defaulted in a tree such that customizing one is known to accelerate dependencies.
 It is open to discussion how we achieve this and that is an independent topic, beyond the scope of this paper.
 
 ## Summary
-Starting with [@P0443R11] as a baseline we have the following customization points:
+Starting with [@P0443R13] as a baseline we have the following customization points:
 
- * `execute(executor, invocable) -> void`
+ * `connect(sender, receiver) -> operation_state`
+ * `start(operation_state) -> void`
  * `submit(sender, receiver) -> void`
  * `schedule(scheduler) -> sender`
+ * `execute(executor, invocable) -> void`
  * `set_done`
  * `set_error`
  * `set_value`
 
 and the following Concepts:
 
- * `executor`
  * `scheduler`
- * `callback_signal`
- * `callback`
+ * `receiver`
+ * `receiver_of`
  * `sender`
+ * `sender_to`
+ * `typed_sender`
+ * `operation_state`
+ * `executor`
+ * `executor_of`
 
 We propose immediately discussing the addition of the following algorithms:
 
  * `just(v)`
    * returns a `sender` of the value `v`
- * `just_via(sch, v)`
-   * a variant of the above that embeds the `via` algorithm
- * `via(s, sch)`
+ * `just_on(sch, v)`
+   * a variant of the above that embeds the `on` algorithm
```

Comment:
I am not clear why just_on appears in the base set and sync_wait_on does not. I would suggest that we need neither now.

Reply:
Diplomacy. I agreed to put it in because Jared thought we needed it.